stdlib/public/runtime/HeapObject.cpp

```diff
@@ -609,13 +610,25 @@ void swift::swift_rootObjCDealloc(HeapObject *self) {
 }
 #endif
 
+#if SWIFT_OBJC_INTEROP
+static bool _check_fast_dealloc() {
+  //This will always be in libobjc, so RTLD_DEFAULT won't have to do an
+  //expensive search in practice
+  return dlsym(RTLD_DEFAULT, "_objc_has_weak_formation_callout") != nullptr;
```

Comment:
We're using `RTLD_NEXT` in a couple of other places, which ensures we don't go off looking at a ton of other dylibs.